include/seqan3/range/view/translate_join.hpp

```diff
@@ -0,0 +1,405 @@
+#include <seqan3/range/detail/random_access_iterator.hpp>
+#include <seqan3/range/view/deep.hpp>
+#include <seqan3/range/view/detail.hpp>
+#include <seqan3/range/view/translate.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+#include <seqan3/range/container/concept.hpp>
+
+namespace seqan3::detail
+{
+
+// ============================================================================
+//  forwards
+// ============================================================================
+
+template <std::ranges::View urng_t>
+//!\cond
+requires std::ranges::SizedRange<urng_t> &&
+         std::ranges::RandomAccessRange<urng_t> &&
+         std::ranges::SizedRange<reference_t<urng_t>> &&
+         std::ranges::RandomAccessRange<reference_t<urng_t>> &&
+         NucleotideAlphabet<reference_t<reference_t<urng_t>>>
+//!\endcond
+class view_translate_join;
+
+// ============================================================================
+//  translate_fn (adaptor definition for both views)
+// ============================================================================
+
+//!\brief Definition of the range adaptor object type for seqan3::view::translate_join.
+struct translate_join_fn
+{
+    //!\brief Store the argument and return a range adaptor closure object.
+    constexpr auto operator()(translation_frames const tf = translation_frames::SIX_FRAME) const
+    {
+        return detail::adaptor_from_functor{*this, tf};
+    }
+
+    /*!\brief            Directly return an instance of the view, initialised with the given parameters.
+     * \param[in] urange The underlying range.
+     * \param[in] tf     The frame that should be used for translation.
+     * \returns          A range of translated sequence(s).
+     */
+    template <std::ranges::Range urng_t>
+    constexpr auto operator()(urng_t && urange, translation_frames const tf = translation_frames::SIX_FRAME) const
+    {
+        static_assert(std::ranges::ViewableRange<urng_t>,
+            "The range parameter to view::translate_join cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::SizedRange<urng_t>,
+            "The range parameter to view::translate_join must model std::ranges::SizedRange.");
+        static_assert(std::ranges::RandomAccessRange<urng_t>,
+            "The range parameter to view::translate_join must model std::ranges::RandomAccessRange.");
+        static_assert(NucleotideAlphabet<reference_t<reference_t<urng_t>>>,
+            "The range parameter to view::translate_join must be over elements of seqan3::NucleotideAlphabet.");
+
+        return detail::view_translate_join{std::forward<urng_t>(urange), tf};
+    }
+
+    //!\brief This adaptor is usable without setting the frames parameter in which case the default is chosen.
+    template <std::ranges::Range urng_t>
+    constexpr friend auto operator|(urng_t && urange, translate_join_fn const & me)
+    {
+        return me(std::forward<urng_t>(urange));
+    }
+};
```

Comment:
I think you can move this below the class and get rid of the forward...

Reply:
What do you mean? The whole adaptor?

Reply:
yep! Usually the adaptor is always below. It just changed places in the translate.hpp because there are multiple views defined inside and one needs the other...